docs/actions-scripts.md

```diff
@@ -1,23 +1,102 @@
+
 However, note the following restrictions and guidance:
 
 * **Write analyses in Python, R, or Stata.**
 You can can use more than one language in a single project if necessary.  You can find more information about the available libraries below.
+
 * **Do not write code that requires an internet connection to run.**
 Any research objects (datasets, libraries, etc.) that are retrieved via the internet should be imported to the repo locally first.
 If this is not possible (for instance if the object size is too large to be transferred via GitHub) then get in touch.
+
 * **Avoid code that consumes a lot of time or memory.** The server is not an infinite resource. We can advise on code optimisation if run-times become problematic.  A good strategy is to split your processing into separate project pipeline actions; the job runner can then choose to run them in parallel if sufficient resources are available.
-* **Write code that runs across different platforms.**
-Since code will be run both locally and within a Linux-based Docker environment. In practice, this just means ensuring you use forward-slashes `/` for directories.
+
+* **Write code that runs in the OpenSAFELY platform.**
+Code will be run within a Linux-based Docker environment. In practice, this just means ensuring you use forward-slashes `/` for directories.
+
 * **Structure your code into discrete chunks, both within scripts, and by splitting into different pipeline actions.**
 This helps with:
 	* readability
 	* bug-finding
 	* parallelisation via the project pipeline
 
 
+## Reading and Writing Outputs
+
+Scripted actions can read and write output files that are persisted in the workspace. These generally fall into two categories: large files of `highly_sensitive` data to feed into other actions, or smaller `moderately_sensitive` outputs intended for review and release.
+
+
+### Large `highly_sensitive` Output Files
+
+It is important that the right files formats are used for large data files. The wrong formats can waste disk space, execution time, and server memory. The specific formats used vary with language ecosystem, but they should always be compressed.
+
+Note: the `cohortextractor` command produces `csv.gz` outputs, which is the current recommended output to be consumed by all supported languages.
+
+
+=== "Python"
+
+    ```python
+    # read compressed csv output from cohortextractor
+    pd.read_csv("output/input.csv.gz)
+
+    # read feather file, decompressed automatically
+    pd.read_feather("output/input.feather")
+
+    # write compressed feather file
+    df.to_feather("output/model.feather", compression="zstd")
+    ```
+
+=== "R"
+
+    ```r
+    # read compressed csv output from cohortextractor
+    df <- readr::read_csv("output/input.csv.gz")
+
+    # read a feather file, decompressed automatically
```

Comment:
can cohortextractor also produce feather files? My reading of this implies that it cannot

Reply:
or is it just for inter-scripted action data (as per .dta.gz in stata)?

Reply:
It can, but we are unsure that feather is ready to be the default just yet.

The intention here was to read feather files in subsequent actions. 

Perhaps I can add switch the order of feather read/write examples with a comment

Reply:
I have switched the order and added a comment for clarity